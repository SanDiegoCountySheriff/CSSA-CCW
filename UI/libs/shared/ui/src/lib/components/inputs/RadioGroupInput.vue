<template>
  <div>
    <v-radio-group
      v-if="layout === 'row'"
      row
      ref="radioGroup"
      v-model="value"
      :label="label"
      :hint="hint"
      persistent-hint
      @change="handleChange"
      @blur="handleBlur"
    >
      <v-radio
        v-for="(option, i) in options"
        :key="i"
        :label="option.label"
        :value="option.value"
      />
    </v-radio-group>

    <v-radio-group
      v-if="layout !== 'row'"
      ref="radioGroup"
      v-model="value"
      :hint="$t(hint)"
      persistent-hint
      :label="$t(label)"
      @change="handleChange"
      @blur="handleBlur"
    >
      <v-radio
        v-for="(option, i) in options"
        :key="i"
        :label="$t(option.label)"
        :value="$t(option.value)"
      />
    </v-radio-group>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import { RadioOptions } from '@shared-ui/types/defualtTypes';

export default defineComponent({
  name: 'RadioGroupInput',
  data: () => ({
    value: '',
  }),
  props: {
    options: {
      type: Array<RadioOptions>,
      default: () => [],
    },
    label: {
      type: String,
      default: '',
    },
    layout: {
      type: String,
      default: '',
    },
    hint: {
      type: String,
      default: '',
    },
    target: {
      type: String,
      default: '',
    },
  },
  methods: {
    handleChange() {
      this.$emit('input', this.value, this.target);
    },
    handleBlur() {
      console.log(this.$refs.radioGroup);

      if (!this.value) {
        console.log(this.$refs.radioGroup);
      }
    },
  },
});
</script>
