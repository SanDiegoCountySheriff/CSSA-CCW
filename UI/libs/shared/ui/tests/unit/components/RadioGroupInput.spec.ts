import RadioGroupInput from '@shared-ui/components/inputs/RadioGroupInput.vue';
import Vuetify from 'vuetify';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('RadioGroupInput', () => {
  let vuetify;

  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = shallowMount(RadioGroupInput, {
      localVue,
      vuetify,
      mocks: tMock,
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it('Should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should Render the correct layout', async () => {
    await wrapper.setProps({ layout: 'row' });
    expect(wrapper.find('v-radio-group-stub').attributes().row).toBeTruthy();
  });

  it('Should render the correct layout non row ', () => {
    expect(wrapper.find('v-radio-group-stub').attributes().row).toBeFalsy();
  });

  it('Should render the correct first options', async () => {
    await wrapper.setProps({
      options: [
        {
          label: 'yes',
          value: 'true',
        },
        {
          label: 'no',
          value: 'false',
        },
      ],
    });
    const options = wrapper.findAll('v-radio-stub');

    expect(options.at(0).attributes().label).toEqual('yes');
    expect(options.at(0).attributes().value).toEqual('true');
  });

  it('Should render the correct first options', async () => {
    await wrapper.setProps({
      options: [
        {
          label: 'yes',
          value: 'true',
        },
        {
          label: 'no',
          value: 'false',
        },
      ],
    });
    const options = wrapper.findAll('v-radio-stub');

    expect(options.at(1).attributes().label).toEqual('no');
    expect(options.at(1).attributes().value).toEqual('false');
  });
});
