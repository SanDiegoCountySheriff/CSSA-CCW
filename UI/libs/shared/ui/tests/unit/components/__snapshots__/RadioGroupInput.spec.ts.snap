// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroupInput Should match snapshot 1`] = `
<div>
  <!---->
  <v-radio-group-stub valuecomparator="[Function]" value="" activeclass="v-item--active" tag="div" errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" height="auto" hint="" label="" persistenthint="true" column="true"></v-radio-group-stub>
</div>
`;
